client/client.go

```diff
@@ -0,0 +1,137 @@
+// Client is a cross-platform client for the signer binary (a.k.a."EnterpriseCertSigner").
+// The signer binary is OS-specific, but exposes a standard set of APIs for the client to use.
+package client
+
+import (
+	"crypto"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/gob"
+	"fmt"
+	"io"
+	"net/rpc"
+	"os"
+	"os/exec"
+
+	"github.com/googleapis/enterprise-certificate-proxy/client/util"
+)
+
+const signAPI = "EnterpriseCertSigner.Sign"
+const certificateChainAPI = "EnterpriseCertSigner.CertificateChain"
+const publicKeyAPI = "EnterpriseCertSigner.Public"
+
+// A Transport wraps a pair of unidirectional streams as an io.ReadWriteCloser.
+type Transport struct {
+	io.ReadCloser
+	io.WriteCloser
+}
+
+// Close closes t's underlying ReadCloser and WriteCloser.
```

Comment:
Can you add comments about when this should be called? I think it is important to have the caller understand when the client can be closed.

Reply:
Done.